### PR TITLE
Edits to Intro, Design, and UCT description

### PR DESCRIPTION
**Pull request by manjugv**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/538/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1116/ for details.

**Comment by shamisp** [comment]
The test fails because of: 
```
00:41:24 /scrap/jenkins/jenkins/jobs/gh-ucx-pr/workspace/contrib/../test/gtest/uct/test_ud.cc:176: Failure
00:41:24 Value of: ack_req_tx_cnt
00:41:24   Actual: 4
00:41:24 Expected: 2
00:41:25 [  FAILED  ] ud/test_ud.ack_req_window/1, where GetParam() = ud/mlx4_0:1 (2477 ms)
00:41:25 [ RUN      ] ud/test_ud.connect_iface_single/0
```
Unrelated error

**Comment by shamisp** [comment]
+1 with the change. @yosefe please take a look.

File: doc/doxygen/design.dox [hunk 1]
```diff
@@ -42,9 +42,9 @@
  * \section UCP
  * UCP implements higher-level protocols that are typically used by message passing (MPI)
  * and PGAS programming models by using lower-level capabilities exposed
- * through the UCT layer. UCP is responsible for the following func-
- * tionality: initialization of the library, selection of transports for
- * communication, message fragmentation, and multi-rail communication.
+ * through the UCT layer. UCP is provides the following functionality: ability to select different transports for
```
Comment (yosefe) [inline doc/doxygen/design.dox]: pls split long lines

**Comment by yosefe** [comment]
:+1: The test failures are unrelated

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1124/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1125/ for details.

**Pull request merged**